static/js/markdown.js

```diff
@@ -225,7 +225,18 @@ function handleStream(streamName) {
     return '<a class="stream" data-stream-id="' + stream.stream_id + '" ' +
         'href="' + href + '"' +
         '>' + '#' + escape(stream.name) + '</a>';
+}
 
+function handleStreamTopic(streamName, topic) {
+    var stream = stream_data.get_sub(streamName);
+    if (stream === undefined || !topic) {
+        return;
+    }
+    var href = window.location.origin + '/#narrow/stream/' + hash_util.encode_stream_name(stream.name);
```

Comment:
I think you don't need the `href` here; a relatively link is what we use on the backend, and what we should render here.

Also, `has_util.by_stream_topic_uri` does all this stuff for you

Reply:
I guess these are both pre-existing problems with the `stream` code that we should fix.